scripts/install-build-tools.sh

```diff
@@ -14,30 +15,216 @@ if (( $EUID != 0 )); then
     SUDO='sudo'
 fi
 
+# install in a custom prefix rather than /usr/local. by default, this
+# chooses "prefix" directory alongside "scripts" directory.
+ROOT=$(cd "$(dirname "$0")"/.. && pwd)
+PREFIX="$(cd "$(dirname "$0")"/.. && pwd)/prefix"
+echo "Will install local dependencies in the following prefix: $PREFIX"
+mkdir -p "$PREFIX"/{bin,lib,include}
+
+# Supporting these versions for buildflow
+PYTHON_VERSIONS=("3.12" "3.11" "3.10")
+ENV_NAME=".py_venv"
+PY_INSTALLED=''
+echo "Python3 versions supported: ${PYTHON_VERSIONS[@]}"
+
+# make a virtual environement to install python packages
+create_venv_install_python() {
+  PY_LOC=$1
+  if [ -z "$PY_LOC" ]; then
+    echo "Python path not provided"
+    exit 1
+  fi
+  PY_VERSION=$2
+  if [ -z "$PY_VERSION" ]; then
+    echo "python version not provided"
+    exit 1
+  fi
+  # remove existing virtual environment if it exists - global var
+  ENV_LOCATION="${ROOT}/${ENV_NAME}"
+  if [ -d "${ENV_LOCATION}" ]; then
+    if rm -rf "${ENV_LOCATION}"; then
```

Comment:
```suggestion
    if rm -rf -- "${ENV_LOCATION}"; then
```

Whenever you use `rm -rf`, *always* pass `--` before any file arguments. The number of times people have been bitten by `-`-containing filenames being interpreted as flags is innumerable, and we should avoid it at all costs.

Reply:
ACK, updated, thanks for the warning